### PR TITLE
refactor(examples): make demos be more consistent

### PR DESCRIPTION
**Pull request by shamilsan**
- Removed unused things.
- Made manifest style united.
- Split contract source files into `lib.rs` + `wasm.rs`.

@gear-tech/dev

### CODE REVIEW

**Comment by StackOverflowExcept1on** [comment]
Can we consider removing `parity-scale-codec` and replacing with `gstd` re-export?

**Comment by shamilsan** [comment]
> Can we consider removing `parity-scale-codec` and replacing with `gstd` re-export?

I don't find importing `parity-scale-codec` too bad.

**Review by clearloop — APPROVED** [review]
finally can remove `workspace = "../../"` lol

File: examples/async-init/Cargo.toml [hunk 1]
```diff
@@ -4,7 +4,6 @@ version = "0.1.0"
 authors.workspace = true
 edition.workspace = true
 license.workspace = true
-workspace = "../../"
```
Comment (breathx) [inline examples/async-init/Cargo.toml:7]: homepage.workspace = true
repository.workspace = true

File: examples/fungible-token/io/Cargo.toml [hunk 2]
```diff
@@ -4,7 +4,6 @@ version = "0.1.4"
 authors.workspace = true
 edition.workspace = true
 license.workspace = true
-workspace = "../../../"
 
 [dependencies]
 gstd = { workspace = true }
```
Comment (breathx) [inline examples/fungible-token/io/Cargo.toml:9]: Seems you didn't totally unify the toml style 
```suggestion
gstd.workspace = true
```

Comment (breathx) [reply examples/fungible-token/io/Cargo.toml:9]: And introduce similar feature gate for all programs
Furthermore you can't split them all into lib.rs and wasm.rs, cause some of our demo doesnt compile correct but don't show up on CI and tests due to cargo workspace issues

**Review by breathx — CHANGES_REQUESTED** [review]

File: examples/fungible-token/io/Cargo.toml (hunk 2 above)
Comment (clearloop) [reply examples/fungible-token/io/Cargo.toml:9]: ref #3069

**Comment by shamilsan** [comment]
@breathx Please re-review

File: examples/compose/src/wasm.rs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// This contract recursively composes itself with another contract (the other contract
```
Comment (breathx) [inline examples/compose/src/wasm.rs:19]: doc comment

File: examples/gas-burned/src/wasm.rs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+use gstd::{msg, prelude::*};
```
Comment (breathx) [inline examples/gas-burned/src/wasm.rs:1]: maybe keep mod wasm inside lib rs for small programs?

**Review by breathx — APPROVED** [review]
Btw you can significantly reduce diff by initially just raneming previous lib.rs into wasm.rs and moving crate-related stuff into lib.rs (it will be the only diff just to keep history of demo authors etc)

File: examples/gas-burned/src/wasm.rs (hunk 4 above)
Comment (shamilsan) [reply examples/gas-burned/src/wasm.rs:1]: I prefer a common approach for all.

**Pull request merged**